exercises/clock/.meta/generator/clock_case.rb

```diff
@@ -9,27 +9,49 @@ def name
   end
 
   def workload
-    property == 'equal' ? compare_clocks : simple_test
+    case property
+    when 'create'
+      simple_test
+    when 'add'
+      add_to_clock
+    when 'subtract'
+      subtract_from_clock
+    when 'equal'
+      compare_clocks
+    else
+      raise "Encountered unknown property in canonical-data.json"
+    end
   end
 
   private
 
   def compare_clocks
-    "clock1 = Clock.at(#{clock1['hour']}, #{clock1['minute']})
-    clock2 = Clock.at(#{clock2['hour']}, #{clock2['minute']})
+    first_hour = input['clock1']['hour']
+    first_minute = input['clock1']['minute']
+    second_hour = input['clock2']['hour']
+    second_minute = input['clock2']['minute']
+
+    "clock1 = Clock.new(hour: #{first_hour}, minute: #{first_minute})
+    clock2 = Clock.new(hour: #{second_hour}, minute: #{second_minute})
     #{assert} clock1 == clock2"
   end
 
   def simple_test
     [
       "assert_equal #{expected.inspect}, ",
-      "#{'(' if add_to_clock}Clock.at(#{hour}, ",
-      "#{minute})#{add_to_clock}#{')' if add_to_clock}.to_s"
+      "Clock.new(#{"hour: #{input['hour']}" unless input['hour'] == 0}",
+      "#{", " unless input['hour'] == 0 || input['minute'] == 0}",
+      "#{"minute: #{input['minute']}" unless input['minute'] == 0}).to_s"
```

Comment:
It's hard to see what is happening here, and those conditionals scare me. What about:
```ruby
    keyword_arguments = %w[hour minute].map { |key| [key, input[key]] }
    keyword_arguments.reject! { |_, value| value.zero? }
    keyword_arguments.map! { |key, value| "#{key}: #{value}" }

    clock_builder = "Clock.new(#{keyword_arguments.join(', ')}).to_s"
    "assert_equal #{expected.inspect}, #{clock_builder}"
  end
```

Reply:
Nice, I didn't really want to have to embed that logic the way it is right now.

I really like your suggestion here.  It took me a minute to get how it works, but it is awesome.

Thank you.